doc/source/whatsnew/v0.22.0.txt

```diff
@@ -205,4 +205,5 @@ Other
 
 - Improved error message when attempting to use a Python keyword as an identifier in a numexpr query (:issue:`18221`)
 - Fixed a bug where creating a Series from an array that contains both tz-naive and tz-aware values will result in a Series whose dtype is tz-aware instead of object (:issue:`16406`)
+- Fixed initialization of Series from dict containing NaN as key (:issue:`18480`)
```

Comment:
intialization -> construction
use double-backticks around ``Series`` and ``NaN`` and ``dict``